internal/worker_runner.go

```diff
@@ -157,49 +206,45 @@ func (wr *workerRunner) queryLocation(modem string) (*models.Location, error) {
 	return &location, nil
 }
 
-func (wr *workerRunner) queryOBD(queryOBD bool, fingerprintDone bool, powerStatus api.PowerStatusResponse, signals []models.SignalData) []models.SignalData {
-	if queryOBD {
-		// do fingerprint but only once
-		if !fingerprintDone {
-			err := wr.fingerprintRunner.FingerprintSimple(powerStatus)
-			if err != nil {
-				wr.logger.Err(err).Msg("failed to do vehicle fingerprint")
-			} else {
-				fingerprintDone = true
+func (wr *workerRunner) queryOBD() {
+	for _, request := range wr.pids.Requests {
+		// check if ok to query this pid
+		if lastEnqueuedTime, ok := wr.signalsQueue.lastEnqueuedTime(request.Name); ok {
+			if int(time.Since(lastEnqueuedTime).Seconds()) < request.IntervalSeconds {
+				continue
 			}
+		} else {
```

Comment:
i don't think we need this, or can make it debug

Reply:
we end up here the very first time, when the device starts(`sudo systemctl start edge-network`) and it tries to query the PID the first time. We can remove it